packages/server/postgres/migrations/1625124096757_addUserEmailUniqueConstraint.ts

```diff
@@ -0,0 +1,18 @@
+import {MigrationBuilder, ColumnDefinitions} from 'node-pg-migrate'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.sql(`
+     ALTER TABLE "User"
+       ADD CONSTRAINT "User_email_key"
+       UNIQUE ("email");
```

Comment:
In postgres, adding a unique constraint is one of many operations that will lock the entire table which is no good. Can you try using the [following workaround](https://leopard.in.ua/2016/09/20/safe-and-unsafe-operations-postgresql#.YN4S-hNKi-8) to get around it?

> As workaround, you can add column, add unique index concurrently, and then add the constraint onto the table:

```
CREATE UNIQUE INDEX CONCURRENTLY token_is_unique ON large_table(token);
ALTER TABLE large_table ADD CONSTRAINT token UNIQUE USING INDEX token_is_unique;
```